src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -30,299 +31,278 @@ export const sortCurrencies = (selectedCurrency, allCurrencies) => {
-    hideCurrencyExpansionControls: PropTypes.bool,
-    defaultExpandCurrencies: (props, propName, componentName, ...rest) => {
-      if (
-        props.hideCurrencyExpansionControls &&
-        typeof props[propName] === 'boolean'
-      ) {
-        throw new Error(
-          oneLine`
-            ${componentName}: "${propName}" does not have any effect when
-            "hideCurrencyExpansionControls" is set.
-          `
-        );
-      }
-      return PropTypes.bool(props, propName, componentName, ...rest);
-    },
-    isDisabled: PropTypes.bool,
-    placeholder: PropTypes.objectOf(PropTypes.string),
-    horizontalConstraint: PropTypes.oneOf(['m', 'l', 'xl', 'scale']),
-    hasError: PropTypes.bool,
-    hasWarning: PropTypes.bool,
-    errors: PropTypes.objectOf(PropTypes.node),
-    warnings: PropTypes.objectOf(PropTypes.node),
-    // HoC
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-  };
+LocalizedInput.propTypes = {
+  id: PropTypes.string,
+  name: PropTypes.string,
+  value: PropTypes.shape({
+    amount: PropTypes.string.isRequired,
+    currencyCode: PropTypes.string.isRequired,
+  }).isRequired,
+  onChange: PropTypes.func,
+  currency: PropTypes.string.isRequired,
+  onBlur: PropTypes.func,
+  onFocus: PropTypes.func,
+  isDisabled: PropTypes.bool,
+  currenciesControl: PropTypes.node,
+  hasError: PropTypes.bool,
+  hasWarning: PropTypes.bool,
+  placeholder: PropTypes.string,
+  error: PropTypes.node,
+  warning: PropTypes.node,
+};
 
-  static getId = getId;
+const LocalizedMoneyInput = props => {
+  const intl = useIntl();
 
-  static getName = getName;
+  const defaultExpansionState =
+    props.hideCurrencyExpansionControls ||
+    props.defaultExpandCurrencies ||
+    // default to `false`, because useToggleState defaults to `true`
+    false;
 
-  static defaultProps = {
-    horizontalConstraint: 'scale',
-  };
+  const [areCurrenciesExpanded, toggleCurrencies] = useToggleState(
+    defaultExpansionState
+  );
 
-  static convertToMoneyValues = (values, locale) =>
-    Object.values(values).map(value =>
-      MoneyInput.convertToMoneyValue(value, locale)
-    );
+  const id = getFieldId(props, {}, sequentialId);
 
-  static parseMoneyValues = (moneyValues = [], locale) =>
-    moneyValues
-      .map(value => MoneyInput.parseMoneyValue(value, locale))
-      .reduce(
-        (pairs, value) => ({
-          ...pairs,
-          [value.currencyCode]: value,
-        }),
-        {}
-      );
+  const hasErrorInRemainingCurrencies =
+    props.hasError ||
+    getHasErrorOnRemainingLanguages(props.errors, props.selectedCurrency);
 
-  static getHighPrecisionCurrencies = (values, locale) =>
-    Object.keys(values).filter(currencyCode =>
-      MoneyInput.isHighPrecision(values[currencyCode], locale)
-    );
+  if (hasErrorInRemainingCurrencies) {
+    if (hasErrorInRemainingCurrencies !== areCurrenciesExpanded) {
+      toggleCurrencies();
```

Comment:
Setting state in render? Would this be more an effect?

Reply:
I added the comment that we added in the localized-text-input

```
    // this update within render replaces the old `getDerivedStateFromProps` functionality
    // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
```

Reply:
"This might look strange at first, but an update during rendering is exactly what getDerivedStateFromProps has always been like conceptually."